src/client/opamSolution.ml

```diff
@@ -1269,19 +1271,20 @@ let install_sys_packages ~map_sysmap ~confirm env config sys_packages t =
     give_up_msg ();
     OpamStd.Sys.exit_because `Aborted
   in
-  if OpamSysPkg.Set.is_empty sys_packages ||
+  if (OpamSysPkg.Set.is_empty sys_packages && OpamSysPkg.Set.is_empty required) ||
      OpamClientConfig.(!r.show) ||
      OpamClientConfig.(!r.assume_depexts) then
     t
   else
   try
     OpamConsole.header_msg "Handling external dependencies";
     OpamConsole.msg "\n";
-    entry_point t sys_packages
+    entry_point t sys_packages required
   with Sys.Break as e -> OpamStd.Exn.finalise e give_up_msg
 
-let install_depexts ?(force_depext=false) ?(confirm=true) t packages =
+let install_depexts ?(force_depext=false) ?(confirm=true) t ~new_packages ~all_packages =
   let map_sysmap f t =
+    let t = Option.get t in
```

Comment:
could you explain why all these `Option.get` are needed?

Reply:
It's because I don't have access to the switch state here https://github.com/ocaml/opam/pull/5982/commits/edd90302e11813b3ac51f37277fb40fd9f3159a9#diff-6426799f367266b2a65325c870931b51df35852e12ddbe8580e74f4eeeb58177R1606 afaik because it hasn't been created at this point. We need to pass it in to `OpamSysInteract.install*` in order for Nix to write to files in the switch, but we also need to support installing system packages before this switch has been created in the existing `OpamClient` logic (which will do nothing for the Nix backend).